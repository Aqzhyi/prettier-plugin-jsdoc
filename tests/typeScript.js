/* eslint-disable no-undef */
const prettier = require("prettier");

function subject(code, options = {}) {
  try {
    return prettier.format(code, {
      parser: "jsdoc-parser",
      plugins: ["."],
      jsdocSpaces: 1,
      ...options,
    });
  } catch (error) {
    console.error(error);
  }
}

test("JS code should be formatted as usuall", () => {
  const result = subject(`
  /**
 @typedef {
    {
        "userId": string,
        "title": string,
        "profileImageLink": string,
        "identityStatus": "None",
        "isBusinessUser": boolean,
        "isResellerUser": boolean,
        "isSubUser": boolean,
        "shareCode": number,
        "referredBy": string,
        "businessName": string,
        "businessUserId": string,
        "nationalCode": string,
        "state": string,
        "city": string,
        "address": string,
        "phoneNumber": string
      }
     } User
     */
  export let User

  /**
     @typedef {
      {
        "domainId": 0,
        persianName: string,
        "englishName": string, // comment
        "resellerUserId": string,
        "isActive": true,
        "logoFileUniqueId": string,
        "logoFileName": string,
        "logoFileUrl": string,
        "domainPersianName": string,
        "domainEnglishName": string,
        "resellerUserDisplayName": string,
        "about": string
      }
     } SubDomain
     */

    /**
     @typedef {
      () => a.b
     } SubDomain
     */
    `);

  expect(result).toMatchSnapshot();
});

test("hoisted object", () => {
  const result = subject(`
  /**
 @typedef {
    {
        "userId": {
        title: string,
        "profileImageLink": *,
        "identityStatus": "None",
        "isBusinessUser": "isResellerUser"|"isBoolean"|  "isSubUser" |    "isNot",
        "shareCode": number,
        "referredBy": any,
        },
        id:number
      }
     } User
     */
 
    `);

  expect(result).toMatchSnapshot();
});
