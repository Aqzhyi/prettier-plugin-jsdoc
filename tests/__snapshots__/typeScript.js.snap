// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JS code should be formatted as usuall 1`] = `
"/**
 * @typedef {{
 *   userId: string;
 *   title: string;
 *   profileImageLink: string;
 *   identityStatus: \\"None\\";
 *   isBusinessUser: boolean;
 *   isResellerUser: boolean;
 *   isSubUser: boolean;
 *   shareCode: number;
 *   referredBy: string;
 *   businessName: string;
 *   businessUserId: string;
 *   nationalCode: string;
 *   state: string;
 *   city: string;
 *   address: string;
 *   phoneNumber: string;
 * }} User
 */
export let User;

/**
 * @typedef {{
 *   domainId: 0;
 *   persianName: string;
 *   englishName: string; // comment
 *   resellerUserId: string;
 *   isActive: true;
 *   logoFileUniqueId: string;
 *   logoFileName: string;
 *   logoFileUrl: string;
 *   domainPersianName: string;
 *   domainEnglishName: string;
 *   resellerUserDisplayName: string;
 *   about: string;
 * }} SubDomain
 */

/**
 * @typedef {() => a.b} SubDomain
 */
"
`;

exports[`hoisted object 1`] = `
"/**
 * @typedef {{
 *   userId: {
 *     title: string;
 *     profileImageLink: any;
 *     identityStatus: \\"None\\";
 *     isBusinessUser: \\"isResellerUser\\" | \\"isBoolean\\" | \\"isSubUser\\" | \\"isNot\\";
 *     shareCode: number;
 *     referredBy: any;
 *   };
 *   id: number;
 * }} User
 */
"
`;
